src/python/zquantum/core/serialization.py

```diff
@@ -3,18 +3,34 @@
 from operator import attrgetter
 from typing import Any, Iterator
 import numpy as np
-from .bitstring_distribution import BitstringDistribution
+from scipy.optimize import OptimizeResult
+
+from .interfaces.optimizer import optimization_result
+from .history.recorder import HistoryEntry, HistoryEntryWithArtifacts
+from .bitstring_distribution import BitstringDistribution, is_bitstring_distribution
 from .utils import convert_array_to_dict, ValueEstimate, SCHEMA_VERSION
 
 
+def has_numerical_keys(dictionary):
+    try:
```

Comment:
This approach is called EAFP, it is outlined e.g. here https://docs.python.org/3/glossary.html

I would typically advocate for using the current approach, but actually, now that you say it I think it is flawed in this particular case because it would also classify strings representing floats as float-like. I fixed it in 091c313